etl/socrata_metadata_pub.py

```diff
@@ -0,0 +1,210 @@
+"""
+Extract metadata about datasets owned and maintained by DTS
+and publishes the data in another dataset.
+"""
+from datetime import datetime
+import pytz
+import logging
+import os
+import json
+import requests
+from requests.auth import HTTPBasicAuth
+
+from sodapy import Socrata
+
+import utils
+
+# Socrata creds
+SO_WEB = os.getenv("SO_WEB")
+SO_TOKEN = os.getenv("SO_TOKEN")
+SO_USER = os.getenv("SO_KEY")
+SO_PASS = os.getenv("SO_SECRET")
+RESOURCE_ID = "28ys-ieqv"
+
+# Unique ID for the data publishing account
+ATD_USER_ID = "8t3r-wq64"
+
+BASE_URL = "https://data.austintexas.gov/d/"
+EDP_URL = "https://datahub.austintexas.gov/d/"
+
+OUT_FIELDS = [
+    "id",
+    "name",
+    "description",
+    "attribution",
+    "type",
+    "updatedAt",
+    "createdAt",
+    "metadata_updated_at",
+    "data_updated_at",
+    "download_count",
+    "publication_date",
+    "page_views_last_week",
+    "page_views_last_month",
+    "page_views_total",
+]
+
+
+def extract():
+    logger.info("Extracting data")
+    basic = HTTPBasicAuth(username=SO_USER, password=SO_PASS)
+
+    # first, get public assets
+    res = requests.get(
+        f"https://api.us.socrata.com/api/catalog/v1?domains=datahub.austintexas.gov&limit=10000",
+    )
+    odp_data = json.loads(res.text)
+
+    # then get datahub assets
+    res = requests.get(
+        f"https://api.us.socrata.com/api/catalog/v1?domains=datahub.austintexas.gov&limit=10000",
+        auth=basic,
+    )
+    edp_data = json.loads(res.text)
+
+    odp_ids = [a["resource"]["id"] for a in odp_data["results"]]
+
+    # Determining if the asset is public or private requires us comparing the two responses
+    data = []
+    for resource in edp_data["results"]:
+        if resource["resource"]["id"] in odp_ids:
+            resource["is_public"] = True
+        else:
+            resource["is_public"] = False
+
+        # Filtering to only DTS account's assets or those with the "Transportation and Mobility" category
+        if "domain_category" not in resource["classification"]:
+            resource["classification"]["domain_category"] = ""
+        if (
+            resource["owner"]["id"] == ATD_USER_ID
+            or resource["classification"]["domain_category"]
+            == "Transportation and Mobility"
+        ):
+            # Filter to not include draft assets
+            if ":" not in resource["resource"]["id"]:
+                data.append(resource)
+    return data
+
+
+def convert_tz(date_str):
+    if not date_str:
+        return None
+
+    # Get UTC timestamp as datetime object
+    utc_datetime = datetime.strptime(date_str, "%Y-%m-%dT%H:%M:%S.%fZ")
+    utc_timezone = pytz.timezone("UTC")
```

Comment:
not necessary at all to revisit this but wanted to mention that i'm excited about the new [ZoneInfo](https://docs.python.org/3/library/zoneinfo.html) library that finally makes it possible to do this with built-ins in python v3.9+